### PR TITLE
nogo: enable cheap SA checks

### PR DESCRIPTION
**Pull request by sluongng**
Some small improvements to nogo setup:

- nogo: ignore generated proto code
- Enable [SA1012](https://staticcheck.dev/docs/checks#SA1012)
- Enable [SA4021](https://staticcheck.dev/docs/checks#SA4021)
- Enable [ST1019](https://staticcheck.dev/docs/checks#ST1019)

**Related issues**: N/A

### CODE REVIEW

File: server/util/flagutil/types/types.go [hunk 1]
```diff
@@ -385,10 +385,18 @@ type FlagAlias[T any] struct {
 func Alias[T any](name string, newNames ...string) *T {
 	f := &FlagAlias[T]{name: name}
 	var flg *flag.Flag
-	for aliaser, ok := common.IsNameAliasing(f), true; ok; aliaser, ok = flg.Value.(common.IsNameAliasing) {
-		if flg = common.DefaultFlagSet.Lookup(aliaser.AliasedName()); flg == nil {
+	aliaser := common.IsNameAliasing(f)
+	ok := true
```
Comment (iain-macdonald) [inline server/util/flagutil/types/types.go:389]: I'd fold the ok setting and checking into the for loop:

```
for ok := true; ok; {
   ...
}
```

**Review by iain-macdonald — CHANGES_REQUESTED** [review]
Looks like //enterprise/server/test/integration/remote_execution/bazel_rbe:bazel_rbe_test is failing on mac with this PR

File: server/util/flagutil/types/types.go (hunk 1 above)
Comment (sluongng) [reply server/util/flagutil/types/types.go]: Good idea. I tried folding it to the top and ended up folding the last statement up there as well.

I think this is quite readable

**Comment by sluongng** [comment]
@iain-macdonald I will revert SA1024 and SA5011 and submit them using separate PR.

The remaining changes are pretty much non-controversial. However because `nogo` currently is attached as part of rules_go's toolchain, updating the config for `nogo` would trigger the recompilation of every package. This surfaces flaky tests and infra failures (that were previously hidden away because of cached runs of our tests).

**Comment by sluongng** [comment]
I will wait for https://github.com/buildbuddy-io/buildbuddy/pull/4341 to land first, then I will rebase this PR to make sure everything works as expected.

**Review by iain-macdonald — APPROVED** [review]

**Pull request merged**